sftp-common.c

```diff
@@ -169,10 +264,28 @@ encode_attrib(struct sshbuf *b, const Attrib *a)
 			return r;
 	}
 	if (a->flags & SSH2_FILEXFER_ATTR_ACMODTIME) {
-		if ((r = sshbuf_put_u32(b, a->atime)) != 0 ||
-		    (r = sshbuf_put_u32(b, a->mtime)) != 0)
+		if ((r = sshbuf_put_u32(b, a->atim.tv_sec)) != 0 ||
```

Comment:
while you're here, shouldn't you do something sensible if the times exceed UINT32_MAX?  That can happen on 64bit time_t platforms (shouldn't be until 2106, but you could get an early start...)

Reply:
Good point, just pushed another commit to do this